app/components/newsPage/cards/common/NewsCardActions.tsx

```diff
@@ -48,12 +63,49 @@ export const NewsCardActions = ({ entry }: NewsCardActionsProps) => {
     }
   }
 
+  async function handleDelete() {
+    try {
+      switch (entry.type) {
+        case ObjectType.POST:
+          deletePost({ postId: entry.item.id });
+          break;
+        case ObjectType.COLLABORATION_COMMENT:
+          deleteProjectCollaborationComment({ commentId: entry.item.id });
+          break;
+        case ObjectType.UPDATE:
+          deleteProjectUpdate(entry.item.id);
+          break;
+        default:
+          break;
+      }
+
+      removeEntry(entry);
+    } catch (error) {
+      console.error(`Error deleting entry of type ${entry.type}:`, error);
+      errorMessage({ message: m.components_newsPage_cards_common_newsCardAction_error() });
+      appInsights.trackException({
+        exception: new Error(`Failed to delete entry of type ${entry.type}`, { cause: error }),
+        severityLevel: SeverityLevel.Error,
+      });
+    }
+  }
+
   return (
     <CardActions sx={cardActionsStyles}>
       <Grid container direction="row" justifyContent="space-between" alignItems="center">
         <Grid item xs={12} sx={footerStyles}>
           <NewsFeedReactionCard entry={entry} />
-          <FollowButtonWithLink isSelected={followedByUser ?? false} entry={entry} onIconClick={toggleFollow} />
+          <Box sx={actionStyles}>
+            {!hideControls && (
+              <>
+                <ResponseControls onResponse={() => respondingInteractions.onStart(entry.item)} />
+                {userIsAuthor && (
```

Comment:
We have a method "displayEditingControls" available in `useNewsCollabCommentCard` hook. We can use it.